contentgrid-spring-audit-logging/src/main/java/com/contentgrid/spring/audit/handler/messaging/AuditEventToCloudEventMessageConverter.java

```
@@ -0,0 +1,94 @@
+
+    @NonNull
+    private final URI source;
+
+    private static final Set<AuditEventToCloudEventMapper> TYPE_MAPPING = Set.of(
+            AuditEventToCloudEventMapper.build(BasicAuditEvent.class)
+                    .cloudEventType("cloud.contentgrid.audit.basic")
+                    .cloudEventSubject(BasicAuditEvent::getRequestUri)
+                    .build(),
+            AuditEventToCloudEventMapper.build(EntityItemAuditEvent.class)
+                    .cloudEventType("cloud.contentgrid.audit.entity",
+                            e -> e.getOperation().name().toLowerCase(Locale.ROOT))
+                    .cloudEventSubject(
+                            e -> e.getOperation() == Operation.CREATE ? e.getResponseLocation() : e.getRequestUri())
+                    .build(),
+            AuditEventToCloudEventMapper.build(AbstractEntityRelationAuditEvent.class)
+                    .cloudEventType("cloud.contentgrid.audit.entity.relation",
+                            e -> e.getOperation().name().toLowerCase(Locale.ROOT))
+                    .cloudEventSubject(AbstractAuditEvent::getRequestUri)
+                    .build(),
+            AuditEventToCloudEventMapper.build(EntityContentAuditEvent.class)
+                    .cloudEventType("cloud.contentgrid.audit.entity.content",
+                            e -> e.getOperation().name().toLowerCase(Locale.ROOT))
+                    .cloudEventSubject(AbstractAuditEvent::getRequestUri)
+                    .build(),
+            AuditEventToCloudEventMapper.build(EntitySearchAuditEvent.class)
+                    .cloudEventType("cloud.contentgrid.audit.entity.list")
+                    .cloudEventSubject(AbstractAuditEvent::getRequestUri)
+                    .build()
+    );
+
+    @Override
+    public Object fromMessage(Message<?> message, Class<?> targetClass) {
+        // TODO: implement converting back from a message to audit event?
```

rather throw an exception, isntead of silently returning `null` ?

---

When returning null, another message converter gets a chance to convert it. If no message converters return an object, an exception is thrown from somewhere.

---

Ok - what should happen with this `// TODO` ? When would we need the conversion in the other direction ?